pkg/workflow/manager.go

```diff
@@ -299,38 +339,139 @@ func (w *WorkflowManager) enqueueController(obj interface{}) {
 }
 
 func (w *WorkflowManager) addJob(obj interface{}) {
+	job := obj.(*k8sBatch.Job)
+	glog.V(3).Infof("addJob %v", job.Name)
+	if workflow := w.getJobWorkflow(job); workflow != nil {
+		w.enqueueController(workflow)
+	}
 }
 
 func (w *WorkflowManager) updateJob(old, cur interface{}) {
-
+	oldJob := old.(*k8sBatch.Job)
+	curJob := cur.(*k8sBatch.Job)
```

Comment:
I don't think it can fail. Somewhere earlier in the process (Reflector if I remember correctly) the type is checked. The only reason we're expecting an interface{} is because the Informer forces us to.

So it's not mandatory, but it might looks strange this way.